daml-foundations/daml-ghc/daml-stdlib-src/DA/Internal/LF.daml

```diff
@@ -97,6 +99,12 @@ instance Show Time where show = primitive @"BEToText"
 instance Eq Time where (==) = primitive @"BEEqual"
 instance Ord Time where (<=) = primitive @"BELessEq"
 
+-- | The `TextMap a` type represents an associative array from keys of type
+-- `Text` to values of type `a`.
+data TextMap a =
+  -- | HIDE
+  TextMap Opaque
```

Comment:
Should Eq/Ord be in this module for it?

Reply:
`Eq` and `Ord` need `TextMap.toList`, which is defined in `DA.TextMap` as `primitive @"BEMapToList"`. It might make sense to add this as `textMapToList` here. I'm not 100% sure though.